lxd-agent/metrics.go

```diff
@@ -372,6 +387,23 @@ func getMemoryMetrics() (metrics.MemoryMetrics, error) {
 		}
 	}
 
+	// Calculate RSS using kernel memory accounting
+	// This is the most accurate and efficient method as it uses the kernel's own memory accounting
+	if foundMemTotal && foundMemFree && foundBuffers && foundCached && foundShmem {
+		// Formula: RSS = MemTotal - (MemFree + Buffers + Cached - Shmem)
+		// This matches how tools like 'free' calculate used memory
+		rssBytes := memTotalBytes - (memFreeBytes + buffersBytes + cachedBytes - shmemBytes)
+		out.RSSBytes = rssBytes
+		logger.Debug("RSS metric using kernel memory accounting", 
+					logger.Ctx{"formula": "MemTotal-(MemFree+Buffers+Cached-Shmem)", "value": rssBytes})
+	} else {
+		// Log warning if we can't calculate RSS using kernel accounting
+		logger.Warn("Kernel RSS calculation failed - omitting RSS metric", 
+				   logger.Ctx{"foundMemTotal": foundMemTotal, "foundMemFree": foundMemFree, 
+							 "foundBuffers": foundBuffers, "foundCached": foundCached, 
+							 "foundShmem": foundShmem})
+	}
```

Comment:
I think we are better off simply leaving `out.RSSBytes` at 0 and not log anything.